Kinvey.Shared/Store/MultiInsertRequest.cs

```diff
@@ -362,19 +362,17 @@ private async Task<KinveyMultiInsertResponse<T>> HandleMultiInsertRequestAsync(I
             }
             catch (KinveyException exception)
             {
-                if (exception.StatusCode == 500)
+                if (exception.StatusCode == 401)
                 {
-                    response.Entities = new List<T>();
-                    response.Errors = new List<Error>();
-                    for (var index = 0; index < entities.Count(); index++)
-                    {
-                        response.Entities.Add(default(T));
-                        response.Errors.Add(new Error { Code = 0, Errmsg = exception.Message, Index = index });
-                    }
+                    throw;
                 }
-                else
+
+                response.Entities = new List<T>();
+                response.Errors = new List<Error>();
+                for (var index = 0; index < entities.Count(); index++)
                 {
-                    throw;
+                    response.Entities.Add(default(T));
+                    response.Errors.Add(new Error { Code = 0, Errmsg = exception.Message, Index = index });
```

Comment:
in this case, the `Index` value should be relative to the batch right? for example let's say the 1st batch works just fine and the 2nd batch fails, in this case the `Index` value should start with `100` right?

Reply:
@heyzooi Yes, it is correct.
We are resolving all indexes according to an initial batch in code implementation below.
https://github.com/Kinvey/dotnet-sdk/blob/master/Kinvey.Shared/Store/MultiInsertRequest.cs#L309

Reply:
oh ok, looks good! Not sure if can be done the same way here, but in Swift we add the `errors` before the `entities` and use the `entities.count` as a offset for the error index, so we don't need to iterate twice just to fix the index. Here's how is done https://github.com/Kinvey/swift-sdk/blob/e4dd8eb1255d6f1e098a40f095d1b81bd416212d/Kinvey/Kinvey/SaveMultiOperation.swift#L293

Reply:
@heyzooi Thank you for the the advice!
I will look into Swift implementation.